src/Datadog.Trace/Configuration/ConfigurationSourceExtensions.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration
+{
+    internal static class ConfigurationSourceExtensions
+    {
+        private const char Splitter = ';';
+
+        public static IEnumerable<string> GetStrings(this IConfigurationSource source, string key)
+        {
+            return source?.GetString(key)
+                ?.Split(new[] { Splitter }, StringSplitOptions.RemoveEmptyEntries)
+                ?? Enumerable.Empty<string>();
+        }
+
+        public static TEnum GetTypedValue<TEnum>(this IConfigurationSource source, string key)
+            where TEnum : struct, IConvertible
+        {
+            Enum.TryParse(source?.GetString(key) ?? "default", ignoreCase: true, out TEnum typedValue);
```

Comment:
Not sure if we want/need the `"default"` here. The zero-values of enums look fine. Is it more performant (I do not think so(?

Reply:
Originally just moved this part. When I'm looking into enum parsing source, the null value is more performant than invalid value. Will fix it.